ruma-api/src/error.rs

```diff
@@ -17,6 +17,14 @@ impl crate::EndpointError for Void {
     }
 }
 
+impl Display for Void {
+    fn fmt(&self, _: &mut fmt::Formatter<'_>) -> fmt::Result {
+        unreachable!()
```

Comment:
No need for a panic call, even if it's most likely going to be optimized out.

```suggestion
        match self {}
```